File: module/Radio/src/Radio/Entity/Show.php
<?php

namespace Radio\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity 
 * @ORM\Table(name="radioshow")
 * */
class Show {

    /**
     * @ORM\Id 
     * @ORM\Column(type="integer") 
     * @ORM\GeneratedValue 
     * */
    protected $id;

    /**
     * @ORM\Column(type="string", length=100) 
     * */
    protected $name;

    /**
     * @ORM\OneToMany(targetEntity="Contribution",mappedBy="show", fetch="EAGER")
     */
    protected $contributors;

    /**
     * @ORM\OneToMany(targetEntity="Scheduling", mappedBy="show")
     */
    protected $schedulings;

    /**
     * @ORM\Column(type="string", length=255, nullable=true) 
     * */
    protected $definition;

    /**
     * @ORM\Column(type="string", length=25,nullable=true) 
     * */
    protected $alias;

    /**
     * @ORM\Column(type="string", length=50,nullable=true) 
     * */
    protected $banner;

    /**
     * @ORM\Column(type="text",nullable=true) 
     * */
    protected $description;

    /**
     * @ORM\Column(type="integer") 
     * */
    protected $type;
    
     /**
     * @ORM\ManyToMany(targetEntity="Url")
     * @ORM\JoinTable(name="show_url",
     *      joinColumns={@ORM\JoinColumn(name="show_id", referencedColumnName="id")},
     *      inverseJoinColumns={@ORM\JoinColumn(name="url_id", referencedColumnName="id", unique=true)}
     *      )
     **/
    protected $urls = array();
    
    public function getUrls() {
        return $this->urls;
    }

    public function setUrls($urls) {
        $this->urls = $urls;
    }
    public function addUrl($url) {
        $this->urls[] = $url;
    }
    
    public function getId() {
        return $this->id;
    }

    public function getName() {
        return $this->name;
    }

    public function getAuthors() {
        return $this->authors;
    }

    public function getDefinition() {
        return $this->definition;
    }

    public function getSlug() {
        return $this->slug;
    }

    public function getBanner() {
        return $this->banner;
    }

    public function getDescription() {
        return $this->description;
    }

    public function getSchedulings() {
        return $this->schedulings;
    }

    public function getAlias() {
        return $this->alias;
    }

    public function toArray() {
        $a = $this->toArrayShort();
        $a['description'] = $this->getDescription();
        return $a;
    }

    public function toArrayShort() {
        $a = array();
        $a['id'] = $this->getId();
        $a['name'] = $this->getName();
        $a['alias'] = $this->getAlias();
        $a['banner'] = $this->getBanner();
        $a['definition'] = $this->getDefinition();
        $a['type'] = $this->getType();
        return $a;
    }

    public function getType() {
        return $this->type;
    }

    public function setType($type) {
        $this->type = $type;
    }
    public function setId($id) {
        $this->id = $id;
    }

    public function setName($name) {
        $this->name = $name;
    }
    public function getContributors() {
        return $this->contributors;
    }

    public function setContributors($contributors) {
        $this->contributors = $contributors;
    }

     
    public function setSchedulings($schedulings) {
        $this->schedulings = $schedulings;
    }

    public function setDefinition($definition) {
        $this->definition = $definition;
    }

    public function setAlias($alias) {
        $this->alias = $alias;
    }

    public function setBanner($banner) {
        $this->banner = $banner;
    }

    public function setDescription($description) {
        $this->description = $description;
    }


}

?>

File: util/model/src/main/java/hu/radio/tilos/model/Show.java
package hu.radio.tilos.model;

import hu.radio.tilos.model.type.ShowStatus;
import hu.radio.tilos.model.type.ShowType;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity()
@Table(name = "radioshow")
public class Show {

    @Id
    private int id;

    @Basic
    @Column
    private String name;

    @Basic
    @Column
    private String alias;

    @Basic
    @Column
    private String definition;

    @Basic
    @Column
    private String description;

    @OneToMany(mappedBy = "show")
    private Set<Mix> mixes = new HashSet();

    @OneToMany(mappedBy = "show")
    private Set<Contribution> contributors = new HashSet();

    @OneToMany(mappedBy = "show")
    private Set<Scheduling> schedulings = new HashSet();

    @Basic
    @Column
    private ShowType type;

    @Basic
    @Column
    private ShowStatus status;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAlias() {
        return alias;
    }

    public void setAlias(String alias) {
        this.alias = alias;
    }

    public String getDefinition() {
        return definition;
    }

    public void setDefinition(String definition) {
        this.definition = definition;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Set<Mix> getMixes() {
        return mixes;
    }

    public void setMixes(Set<Mix> mixes) {
        this.mixes = mixes;
    }

    public Set<Contribution> getContributors() {
        return contributors;
    }

    public void setContributors(Set<Contribution> contributors) {
        this.contributors = contributors;
    }

    public ShowType getType() {
        return type;
    }

    public void setType(ShowType type) {
        this.type = type;
    }

    public ShowStatus getStatus() {
        return status;
    }

    public void setStatus(ShowStatus status) {
        this.status = status;
    }

    public Set<Scheduling> getSchedulings() {
        return schedulings;
    }

    public void setSchedulings(Set<Scheduling> schedulings) {
        this.schedulings = schedulings;
    }
}
